code/game/objects/items/storage/storage.dm

```diff
@@ -383,69 +384,94 @@
 
 	return TRUE
 
+//This proc handles the delay associated with a storage object.
+//If there is no delay, or the delay is negative, it simply returns TRUE.
+//Should return true if the access delay is completed successfully.
+/obj/item/storage/proc/handle_access_delay(obj/item/accessed, mob/user, taking_out = TRUE, alert_user = TRUE)
+	if(!access_delay || !should_access_delay(accessed, user, taking_out))
+		return TRUE
+
+	if(!alert_user)
+		return do_after(user, access_delay, TRUE, src, ignore_turf_checks=TRUE)
+
+	to_chat(user, "<span class='notice'>You begin to [taking_out ? "take" : "put"] [accessed] [taking_out ? "out of" : "into"] [src]")
+	if(!do_after(user, access_delay, TRUE, src, ignore_turf_checks=TRUE))
+		to_chat(user, "<span class='warning'>You fumble [accessed]!</span>")
+		return FALSE
+	return TRUE
+
+//This proc checks to see if we should actually delay access in this scenario.
+//This proc should return TRUE or FALSE
+/obj/item/storage/proc/should_access_delay(obj/item/accessed, mob/user, taking_out)
```

Comment:
Can't do that reliably because a lot of things that shouldnt be subtypes of backpacks are subtypes of backpacks.

Reply:
You can absolutly make it reliable like that, especially since you already have a access_var to discriminate.

The fact that you need both access_var and this proc to be override for this to have an effect is kinda bad